src/cargo/ops/cargo_package/mod.rs

```diff
@@ -417,6 +417,11 @@ fn local_deps<T>(packages: impl Iterator<Item = (Package, T)>) -> LocalDependenc
                 continue;
             };
 
+            // We don't care about cycles
+            if dep.source_id() == pkg.package_id().source_id() {
```

Comment:
It is a bit unintuitive, as comparing source_id doesn't generally imply they are cyclic, but given we filter in local packages earlier it makes sense.

Reply:
Yeah, in general this code assumes that the `source_id` is unique and looks up a package id by the source id.    I could move this into the following `if` so we can then compare package ids.  That won't make a practical differende but maybe it makes the intent clearer?